homeassistant/components/geniushub/__init__.py

```diff
@@ -106,18 +118,102 @@
+            hass_data.pop(CONF_HOST), **hass_data, session=async_get_clientsession(hass)
+        )
+    else:
+        source = "Cloud: "
+        URL = "version"
+        service = GeniusService(
+            hass_data.pop(CONF_TOKEN),
+            **hass_data,
+            session=async_get_clientsession(hass),
+        )
+
+    response = await service.request("GET", URL)
+
+    _LOGGER.debug("__init__.py:validate_input: ", extra=hass_data_in)
+    if source == "Local: ":
+        mac = response["data"]["UID"]
+
+    return {"title": source + mac}
+
+
+async def _async_import(hass: HomeAssistant, base_config: ConfigType) -> None:
+    """Import a config entry from configuration.yaml."""
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": config_entries.SOURCE_IMPORT},
+        data=base_config[DOMAIN],
+    )
+    if result["type"] == FlowResultType.CREATE_ENTRY:
+        async_create_issue(
+            hass,
+            HOMEASSISTANT_DOMAIN,
+            f"deprecated_yaml_{DOMAIN}",
+            breaks_in_ha_version="2024.12.0",
+            is_fixable=False,
+            issue_domain=DOMAIN,
+            severity=IssueSeverity.WARNING,
+            translation_key="deprecated_yaml",
+            translation_placeholders={
+                "domain": DOMAIN,
+                "integration_title": "Genius Hub",
+            },
+        )
+        return
+    async_create_issue(
+        hass,
+        DOMAIN,
+        f"deprecated_yaml_import_issue_{result['reason']}",
+        breaks_in_ha_version="2024.12.0",
+        is_fixable=False,
+        issue_domain=DOMAIN,
+        severity=IssueSeverity.WARNING,
+        translation_key=f"deprecated_yaml_import_issue_{result['reason']}",
+        translation_placeholders={
+            "domain": DOMAIN,
+            "integration_title": "Genius Hub",
+        },
+    )
+
+
+async def async_setup(hass: HomeAssistant, base_config: ConfigType) -> bool:
+    """Set up a Genius Hub system."""
+    if DOMAIN in base_config:
+        hass.async_create_task(_async_import(hass, base_config))
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, config: ConfigEntry) -> bool:
     """Create a Genius Hub system."""
-    hass.data[DOMAIN] = {}
+    hass.data.setdefault(DOMAIN, {})
+    hass_data = dict(config.data)
+    hass.data[DOMAIN][config.entry_id] = hass_data
 
-    kwargs = dict(config[DOMAIN])
-    if CONF_HOST in kwargs:
-        args = (kwargs.pop(CONF_HOST),)
+    # Create a Genius Hub client and broker
+    if CONF_HOST in hass_data:
+        args = (hass_data.pop(CONF_HOST),)
     else:
-        args = (kwargs.pop(CONF_TOKEN),)
-    hub_uid = kwargs.pop(CONF_MAC, None)
+        args = (hass_data.pop(CONF_TOKEN),)
+
+    hub_uid = hass_data.pop(CONF_MAC, None)
 
-    client = GeniusHub(*args, **kwargs, session=async_get_clientsession(hass))
+    client = GeniusHub(*args, **hass_data, session=async_get_clientsession(hass))
```

Comment:
We should check if we can connect to the bridge before setting up the integration